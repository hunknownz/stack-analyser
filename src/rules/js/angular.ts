import { register } from '../../rules.js';

register({
  tech: 'angular',
  files: ['angular.json'],
  dependencies: [
    { type: 'npm', name: 'angular' },
    { type: 'npm', name: /^@angular\//, example: '@angular/forms' },
  ],
});
