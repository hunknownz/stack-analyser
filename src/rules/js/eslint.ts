import { register } from '../../register.js';

register({
  tech: 'eslint',
  name: 'Eslint',
  type: 'tool',
  files: [
    '.eslintrc',
    '.eslintrc.cjs',
    '.eslintrc.json',
    '.eslintrc.js',
    'eslint.config.js',
    'eslint.config.cjs',
    'eslint.config.mjs',
  ],
  dependencies: [{ type: 'npm', name: 'eslint' }],
});
