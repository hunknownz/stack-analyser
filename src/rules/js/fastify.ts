import { register } from '../../register.js';

register({
  tech: 'fastify',
  name: 'Fastify',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'fastify' }],
});
