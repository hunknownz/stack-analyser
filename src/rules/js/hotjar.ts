import { register } from '../../register.js';

register({
  tech: 'hotjar',
  name: 'Hotjar',
  type: 'tool',
  dependencies: [{ type: 'npm', name: '@hotjar/browser' }],
});
