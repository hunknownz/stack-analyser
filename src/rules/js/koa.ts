import { register } from '../../register.js';

register({
  tech: 'koa',
  name: 'Koa',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'koa' }],
});
