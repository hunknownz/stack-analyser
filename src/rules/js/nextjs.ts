import { register } from '../../register.js';

register({
  tech: 'nextjs',
  name: 'Next.js',
  type: 'framework',
  files: ['next.config.js'],
  dependencies: [{ type: 'npm', name: 'next' }],
});
