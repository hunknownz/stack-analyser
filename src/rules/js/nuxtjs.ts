import { register } from '../../register.js';

register({
  tech: 'nuxtjs',
  name: 'Nuxt.js',
  type: 'framework',
  files: ['nuxt.config.js'],
  dependencies: [{ type: 'npm', name: 'nuxt' }],
});
