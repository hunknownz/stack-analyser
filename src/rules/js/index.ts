import './angular.js';
import './bootstrap.js';
import './esbuild.js';
import './eslint.js';
import './express.js';
import './fastify.js';
import './hotjar.js';
import './koa.js';
import './prettier.js';
import './prisma.js';
import './react.js';
import './reactEmail.js';
import './rollup.js';
import './sequelize.js';
import './socketio.js';
import './storybook.js';
import './strapi.js';
import './stylelint.js';
import './tailwind.js';
import './typescript.js';
import './vite.js';
import './vue.js';
import './webpack.js';
