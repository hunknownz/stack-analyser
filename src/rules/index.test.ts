import { describe, expect, it } from 'vitest';

import { registeredTech } from '../register.js';

import '../autoload.js';

describe('all', () => {
  it('should match everything', async () => {
    expect(Array.from(registeredTech.values()).sort()).toMatchSnapshot();
  });
});
