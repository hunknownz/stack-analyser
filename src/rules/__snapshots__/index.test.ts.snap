// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`all > should match everything 1`] = `
[
  "adminer",
  "adobe",
  "adobecommercecloud",
  "adyen",
  "airbyte",
  "airtable",
  "algolia",
  "alibabacloud",
  "alpinejs",
  "amplitude",
  "angular",
  "ansible",
  "anthropic",
  "apache_airflow",
  "apache_cassandra",
  "apache_couchdb",
  "apache_flink",
  "apache_hadoop",
  "apache_hive",
  "apache_iceberg",
  "apache_kafka",
  "apache_solr",
  "apache_spark",
  "apache_storm",
  "apache_thrift",
  "apiplatform",
  "appveyor",
  "assemble",
  "astro",
  "atlasgo",
  "atlassian",
  "atlassian.bitbucket",
  "atlassian.confluence",
  "atlassian.jira",
  "atlassian.opsgenie",
  "atlassian.trello",
  "auth0",
  "aws",
  "aws.amplifyhosting",
  "aws.apigateway",
  "aws.athena",
  "aws.bedrock",
  "aws.cloudformation",
  "aws.cloudfront",
  "aws.cloudsearch",
  "aws.cloudwatch",
  "aws.codebuild",
  "aws.documentdb",
  "aws.dynamodb",
  "aws.ebs",
  "aws.ec2",
  "aws.ecr",
  "aws.ecs",
  "aws.efs",
  "aws.eks",
  "aws.elasticache",
  "aws.fargate",
  "aws.glacier",
  "aws.glue",
  "aws.kafka",
  "aws.kinesis",
  "aws.kms",
  "aws.lambda",
  "aws.lightsail",
  "aws.memorydb",
  "aws.mq",
  "aws.neptune",
  "aws.opensearch",
  "aws.polly",
  "aws.rds",
  "aws.redshift",
  "aws.rekognition",
  "aws.s3",
  "aws.sagemaker",
  "aws.secretsmanager",
  "aws.ses",
  "aws.sfn",
  "aws.sns",
  "aws.sqs",
  "aws.timestream",
  "aws.translate",
  "azure",
  "azure.aks",
  "azure.ci",
  "azure.cosmosdb",
  "azure.functions",
  "azure.mariadb",
  "azure.mysql",
  "azure.openai",
  "azure.postgres",
  "azure.redis",
  "azure.sql",
  "azure.staticwebapps",
  "azure.storage",
  "bash",
  "betterauth",
  "betterstack",
  "bigcommerce",
  "blackfire",
  "blitzjs",
  "bootstrap",
  "box",
  "brevo",
  "browerbase",
  "browserstack",
  "bytebase",
  "c",
  "caddy",
  "celery",
  "chargebee",
  "checkov",
  "chef",
  "circleci",
  "cirrusci",
  "clerk",
  "clickhouse",
  "clicksend",
  "cloudbees",
  "cloudbees.codeship",
  "cloudflare",
  "cloudflare.pages",
  "cloudflare.workers",
  "cockroachdb",
  "codeclimate",
  "codecov",
  "codesandboxci",
  "cohereai",
  "commercetools",
  "consul",
  "contenful",
  "convexdb",
  "couchbase",
  "coveralls",
  "cplusplus",
  "cratedb",
  "crowdin",
  "csharp",
  "css",
  "cypressci",
  "d3js",
  "dart",
  "databricks",
  "datadog",
  "dataiku",
  "datastax",
  "datocms",
  "deepseek",
  "deepsource",
  "deferrun",
  "deno",
  "denodeploy",
  "dependabot",
  "depotdev",
  "digitalocean",
  "discord",
  "discourse",
  "docker",
  "doctrinephp",
  "docusaurus",
  "docusign",
  "droneci",
  "dropbox",
  "drupal",
  "duckdb",
  "dynatrace",
  "elasticcloud",
  "elasticsearch",
  "elasticstack",
  "electron",
  "eleventy",
  "elixir",
  "emberjs",
  "equinix",
  "esbuild",
  "eslint",
  "expodev",
  "expojs",
  "express",
  "fabric",
  "facebook",
  "fastify",
  "fastly",
  "figma",
  "firebase",
  "firebase.firestore",
  "firecrawl",
  "flyio",
  "flyway",
  "gatsby",
  "gcp",
  "gcp.aiplatform",
  "gcp.appengine",
  "gcp.artifactregistry",
  "gcp.bigquery",
  "gcp.bigtable",
  "gcp.cloudbuild",
  "gcp.cloudrun",
  "gcp.containerregistry",
  "gcp.dataflow",
  "gcp.dataproc",
  "gcp.datastore",
  "gcp.dialogflow",
  "gcp.dns",
  "gcp.functions",
  "gcp.gce",
  "gcp.gcs",
  "gcp.gke",
  "gcp.kms",
  "gcp.language",
  "gcp.logging",
  "gcp.maps",
  "gcp.memorystore",
  "gcp.pubsub",
  "gcp.secretmanager",
  "gcp.spanner",
  "gcp.speech",
  "gcp.sql",
  "gcp.tasks",
  "gcp.translate",
  "gcp.vision",
  "geminiai",
  "ghost",
  "gitbook",
  "gitguardian",
  "github",
  "github.actions",
  "github.codeql",
  "github.pages",
  "gitlab",
  "gitlab.ci",
  "glsl",
  "goacmelego",
  "golang",
  "golangcilint",
  "googleanalytics",
  "gradio",
  "grafana",
  "gridsome",
  "groq",
  "haproxy",
  "hashicorp_vault",
  "healthchecksio",
  "helm",
  "heroku",
  "hexojs",
  "hotjar",
  "httpd",
  "hubspot",
  "huggingface",
  "hugo",
  "hyperdx",
  "hypertune",
  "ibmcloud",
  "iftt",
  "infisical",
  "influxdb",
  "inngest",
  "intercom",
  "intuit",
  "java",
  "javascript",
  "jekyll",
  "jenkins",
  "jest",
  "joomla",
  "k6",
  "kentico",
  "kibana",
  "klarna",
  "klaviyo",
  "koa",
  "koalaanalytics",
  "kong",
  "koyeb",
  "kubernetes",
  "lago",
  "laravel",
  "launchdarkly",
  "lemonsqueezy",
  "libsql",
  "lighthouse",
  "linear",
  "liquibase",
  "logrocket",
  "logsnag",
  "logstash",
  "magento",
  "mailchimp",
  "mailgun",
  "mailhog",
  "mailjet",
  "mariadb",
  "matomo",
  "mattermost",
  "meilisearch",
  "meilisearchcloud",
  "memcached",
  "meteorjs",
  "milvusdb",
  "mintlify",
  "mistralai",
  "mithriljs",
  "mixpanel",
  "mongodb",
  "mongodbatlas",
  "mongoexpress",
  "mysql",
  "n8n",
  "nango",
  "nats",
  "neo4j",
  "neondb",
  "nestjs",
  "netlify",
  "newrelic",
  "nextcloud",
  "nextjs",
  "nginx",
  "nodejs",
  "notion",
  "novu",
  "nuxtjs",
  "nxcloud",
  "nxjs",
  "oceanbase",
  "okta",
  "onlineornot",
  "openai",
  "optimizely",
  "oraclecloud",
  "ovh",
  "ovh.database",
  "ovh.dedicated",
  "ovh.kubernetes",
  "ovh.storage",
  "ovh.vps",
  "paddle",
  "pagerduty",
  "papertrail",
  "payloadcms",
  "paypal",
  "percona",
  "perplexityai",
  "phaserjs",
  "php",
  "phppest",
  "phpstan",
  "phpunit",
  "pingdom",
  "pirschanalytics",
  "placekit",
  "planetscale",
  "platformsh",
  "plausible",
  "playwright",
  "pnpm",
  "postgresql",
  "postgrest",
  "posthog",
  "postman",
  "postmark",
  "preactjs",
  "prestashop",
  "prettier",
  "prisma",
  "prismacloud",
  "prometheus",
  "puppeteer",
  "python",
  "pytorch",
  "qdrant",
  "qovery",
  "qovery.cluster",
  "qovery.database",
  "questdb",
  "rabbitmq",
  "radixui",
  "railway",
  "railway.mongodb",
  "railway.mysql",
  "railway.postgres",
  "railway.redis",
  "react",
  "reactemail",
  "redis",
  "redwoodjs",
  "refinedev",
  "relativeci",
  "remixrun",
  "render",
  "renovate",
  "replit",
  "replit.database",
  "replit.postgres",
  "resend",
  "rethinkdb",
  "rollbar",
  "rollup",
  "ruby",
  "rust",
  "salesforce",
  "sanity",
  "sap",
  "sas",
  "scaleway",
  "scaleway.container",
  "scaleway.database",
  "scaleway.documentdb",
  "scaleway.elasticmetal",
  "scaleway.function",
  "scaleway.kubernetes",
  "scaleway.mq",
  "scaleway.redis",
  "scaleway.secretmanager",
  "scaleway.storage",
  "scoutapm",
  "scss",
  "selenium",
  "sendgrid",
  "sentry",
  "sequelize",
  "servicenow",
  "shopify",
  "shopify.hydrogen",
  "signoz",
  "sitecore",
  "sitecore.xmlcloud",
  "slack",
  "snowflake",
  "snyk",
  "socketio",
  "solidjs",
  "sonarcloud",
  "sonarlint",
  "sonarqube",
  "splitio",
  "splunk",
  "sqlite",
  "sqreen",
  "squarespace",
  "squareup",
  "storybook",
  "strapi",
  "stripe",
  "styleci",
  "stylelint",
  "supabase",
  "supabase.auth",
  "supabase.functions",
  "supabase.postgres",
  "supabase.realtime",
  "supabase.storage",
  "surrealdb",
  "sveltejs",
  "symfony",
  "tailscale",
  "tailwind",
  "tanstackstart",
  "tdengine",
  "teamcity",
  "teamspeak",
  "telegraf",
  "telegram",
  "tencentcloud",
  "tensorflow",
  "terraform",
  "terragrunt",
  "tidb",
  "timescaledb",
  "tinybird",
  "tiptap",
  "traefik",
  "travisci",
  "trufflesecurity",
  "turborepo",
  "tursodb",
  "twigphp",
  "twilio",
  "twitter",
  "typescript",
  "typesense",
  "typesensecloud",
  "upstash",
  "upstash.kafka",
  "upstash.qstash",
  "upstash.redis",
  "uptimekuma",
  "vale",
  "vercel",
  "vercel.ai",
  "vercel.analytics",
  "vercel.blob",
  "vercel.edge",
  "vercel.kv",
  "vercel.postgres",
  "veriff",
  "vialink",
  "victoriametrics",
  "vite",
  "vitest",
  "vue",
  "wasp",
  "webflow",
  "webpack",
  "wiz",
  "woocommerce",
  "wordpress",
  "workos",
  "xai",
  "yarn",
  "yii2",
  "yousign",
  "zapier",
  "zendesk",
  "zig",
  "zipkin",
  "zookeeper",
  "zoom",
  "zuora",
]
`;
